telemetry/internal/appinsights/internal/contracts/envelope.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package contracts
+
+import (
+	"errors"
+	"time"
+)
+
+// Data struct to contain both B and C sections.
+type Data struct {
+	BaseType string    `json:"baseType"`
+	BaseData EventData `json:"baseData"`
+}
+
+// Instances of Event represent structured event records that can be grouped
+// and searched by their properties. Event data item also creates a metric of
+// event count by name.
+type EventData struct {
+	// Schema version
+	Ver int `json:"ver"`
+
+	// Event name. Keep it low cardinality to allow proper grouping and useful
+	// metrics.
+	Name string `json:"name"`
+}
+
+// System variables for a telemetry item.
+type Envelope struct {
+	// Envelope version. For internal use only. By assigning this the default, it
+	// will not be serialized within the payload unless changed to a value other
+	// than #1.
+	Ver int `json:"ver"`
+
+	// Type name of telemetry data item.
+	Name string `json:"name"`
+
+	// Event date time when telemetry item was created. This is the wall clock
+	// time on the client when the event was generated. There is no guarantee that
+	// the client's time is accurate. This field must be formatted in UTC ISO 8601
+	// format, with a trailing 'Z' character, as described publicly on
+	// https://en.wikipedia.org/wiki/ISO_8601#UTC. Note: the number of decimal
+	// seconds digits provided are variable (and unspecified). Consumers should
+	// handle this, i.e. managed code consumers should not use format 'O' for
+	// parsing as it specifies a fixed length. Example:
+	// 2009-06-15T13:45:30.0000000Z.
+	Time time.Time `json:"time"`
+
+	// Sampling rate used in application. This telemetry item represents 1 /
+	// sampleRate actual telemetry items.
+	SampleRate float64 `json:"sampleRate"`
```

Comment:
This is confusing... this telemetry item isn't an "actual" telemetry item?

(Also, more non-go-idiom starts of comments throughout the file.)

Reply:
I've fixed the non-idiomatic comments. About the comment, this comes directly from the API documentation, and is the same in other sdks, so I wouldn't touch it. The phrase is confusing, but I think it refers to the fact that a single envelope can be the result of sampling multiple items. In fact I might be able to use this to implement `Event.Add` where `n` is higher than 1. I'll give it a try once I can upload telemetry to a dev appinishgts endpoint.